frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+    ).then(
+      () => {
+        this.setState({inProgress: false});
+        if (metadata.namespace) {
+          this.props.setActiveNamespace(metadata.namespace);
+        }
+        history.push(formatNamespacedRouteForResource('secrets'));
+      },
+      err => this.setState({error: err.message, inProgress: false})
+    );
+  }
+  render () {
+    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretType)} Secret`;
+    const { saveButtonText } = this.props;
+
+    let explanation = 'Webhook secret allow you to authenticate a webhook trigger.';
+    let subform = <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
+
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group" onSubmit={this.save}>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">{explanation}</p>
+
+        <fieldset disabled={!this.props.isCreate}>
+          <div className="form-group">
+            <label className="control-label">Secret Name</label>
+            <div>
+              <input className="form-control" type="text" onChange={this.onNameChanged} value={this.state.secret.metadata.name} required id="test--subject-name" />
+              <p className="help-block">Unique name of the new secret.</p>
+            </div>
+          </div>
+        </fieldset>
+        {subform}
+        <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress} >
+          <button type="submit" className="btn btn-primary" id="create-secret">{saveButtonText || 'Create'}</button>
+          <Link to={formatNamespacedRouteForResource('secrets')} className="btn btn-default">Cancel</Link>
+        </ButtonBar>
+      </form>
+    </div>;
+  }
+}
+
+const BaseEditSecret = connect(null, {setActiveNamespace: UIActions.setActiveNamespace})(
+  (props: BaseEditSecretProps_) => <BaseEditSecret_ {...props} />
+);
+
+const BindingLoadingWrapper = props => {
+  const fixed = _.reduce(props.fixedKeys, (acc, k) => ({...acc, k: _.get(props.obj.data, k)}), {});
+  return <StatusBox {...props.obj}>
+    <BaseEditSecret {...props} obj={props.obj.data} fixed={fixed} />
+  </StatusBox>;
+};
+
+export const CreateSecret = ({match: {params}}) => {
+  return <BaseEditSecret
+    fixed={{ metadata: {namespace: params.ns} }}
+    metadata={{ namespace: getActiveNamespace() }}
+    secretType={params.type}
+    titleVerb="Create"
+    isCreate={true}
+  />;
+};
+
+export const EditSecret = ({match: {params}, kind}) => <Firehose resources={[{kind: kind, name: params.name, namespace: params.ns, isList: false, prop: 'obj'}]}>
+  <BindingLoadingWrapper fixedKeys={['kind', 'metadata']} titleVerb="Edit" saveButtonText="Save Changes" />
+</Firehose>;
+
+/* eslint-disable no-undef */
+export type BaseEditSecretState_ = {
```

Comment:
Is the trailing underscore common?  I haven't seen previously.

Reply:
It is, at least we use it in our [codebase](https://github.com/openshift/console/blob/master/frontend/public/components/RBAC/bindings.jsx#L343)

Reply:
Huh, ok neat.

Reply:
Not typically for the types, though?

Reply:
Probably not. If anything we could rename:
- `BaseEditSecret`    ->  `BaseEditSecretWrapper`
- `BaseEditSecret_`  ->  `BaseEditSecret`

Reply:
I think the names are good. Just wondering if we should have the underscore on the type definitions